include/galsim/SBInterpolatedImage.h

```diff
@@ -248,20 +146,75 @@ namespace galsim {
         void calculateMaxK(double max_maxk=0.) const;
 
         ConstImageView<double> getImage() const;
-        boost::shared_ptr<Interpolant> getXInterp() const;
-        boost::shared_ptr<Interpolant> getKInterp() const;
 
     protected:
 
         class SBInterpolatedImageImpl;
 
-        // Regular SBProfile pimpl constructor so as to be available to derived classes
-        SBInterpolatedImage(SBProfileImpl* pimpl) : SBProfile(pimpl) {}
+        // Wow, three classes here... I'm not even sure how this works,
+        // but the tests seem to pass...
```

Comment:
(1) This comment does not exactly inspire confidence.  :)
(2) Honestly, I agree.  I find this confusing.  If anyone wants to educate me more on this, I'm happy to learn.